pkg/crosscluster/logical/sql_row_writer.go

```diff
@@ -0,0 +1,175 @@
+// Copyright 2025 The Cockroach Authors.
+//
+// Use of this software is governed by the CockroachDB Software License
+// included in the /LICENSE file.
+
+package logical
+
+import (
+	"context"
+
+	"github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl/cdcevent"
+	"github.com/cockroachdb/cockroach/pkg/sql/catalog"
+	"github.com/cockroachdb/cockroach/pkg/sql/isql"
+	"github.com/cockroachdb/cockroach/pkg/sql/parser/statements"
+	"github.com/cockroachdb/cockroach/pkg/sql/sem/tree"
+	"github.com/cockroachdb/cockroach/pkg/sql/sessiondata"
+	"github.com/cockroachdb/cockroach/pkg/sql/types"
+	"github.com/cockroachdb/errors"
+)
+
+// errStalePreviousValue is returned if the row supplied to UpdateRow,
+// UpdateTombstone, or DeleteRow does not match the value in the local
+// database.
+var errStalePreviousValue = errors.New("stale previous value")
+
+// sqlRowWriter is configured to write rows to a specific table and descriptor
+// version.
+type sqlRowWriter struct {
+	insert statements.Statement[tree.Statement]
```

Comment:
what's the value of removing the queryBuilder abstraction, which contains a map of tableDescID to sql stmts? Perhaps this will become more apparent when i see the new row processor.